pageserver/src/import_datadir.rs

```diff
@@ -617,3 +619,27 @@ async fn read_all_bytes(reader: &mut (impl AsyncRead + Unpin)) -> Result<Bytes>
     reader.read_to_end(&mut buf).await?;
     Ok(Bytes::from(buf))
 }
+
+pub async fn create_tar_zst(pgdata_path: &Path) -> Result<Vec<u8>> {
+    let mut paths = Vec::new();
+    for entry in WalkDir::new(pgdata_path) {
+        let entry = entry?;
+        let metadata = entry.metadata().expect("error getting dir entry metadata");
```

Comment:
bootstrap_timeline was cancellable for a long time, so I think it would handle `?` just fine, unsure about panicking. Did you check this?

Reply:
This does what `import_timeline_from_postgres_datadir` does. Not sure what you mean by `bootstrap_timeline` being cancellable.

Reply:
Before earlier this summer, we cancelled for example bootstrap_timeline if hyper dropped the future (when client closed the connection). It is also possible that we just did not have any await points there to be dropped at back then.

Now a task is spawned for the request handling. I was wondering if there could be some `if let Err(e) = res` branches which did some some cleanup, which would not get executed on panic, but perhaps these are rare enough.

Reply:
Okay so you want me to add a cancellation token? Sure I can do that.

Reply:
No cancellation token, I was asking about the panicking.

Reply:
What I was remembering was the scopeguard::defer in the caller so this can now be resolved.